### PR TITLE
Use Lock mechanism for forward_model_ok

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
Resolves https://github.com/equinor/ert/issues/8070

**Approach**
Make sure that we will not run more than 1 internalization job at a time. 

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/scheduler.py [hunk 1]
```diff
@@ -275,6 +275,7 @@ async def execute(
             scheduling_tasks.append(asyncio.create_task(self._update_avg_job_runtime()))
 
         sem = asyncio.BoundedSemaphore(self._max_running or len(self._jobs))
+        self._fmok_sem = asyncio.BoundedSemaphore(value=10)
```
Comment (berland) [inline src/ert/scheduler/scheduler.py:278]: `value=1` should be enough for everybody

Comment (xjules) [reply src/ert/scheduler/scheduler.py]: Actually, when it is only `1` I can use lock instead :+1:

File: src/ert/scheduler/scheduler.py [hunk 2]
```diff
@@ -275,6 +276,9 @@ async def execute(
             scheduling_tasks.append(asyncio.create_task(self._update_avg_job_runtime()))
 
         sem = asyncio.BoundedSemaphore(self._max_running or len(self._jobs))
+        # this semaphore is to assure that no more than 1 task
+        # does internationalization at a time
```
Comment (berland) [inline src/ert/scheduler/scheduler.py:280]: internalization (not being international today)

**Comment by xjules** [comment]
Tested on bigpoly (300 realz, max_running 200) both on LSF and LOCAL queues and it worked.

File: src/ert/scheduler/job.py [hunk 3]
```diff
@@ -223,17 +223,25 @@ async def _verify_checksum(self, timeout: int = 120) -> None:  # noqa: ASYNC109
...
-            self._callback_status_msg = status_msg
-        else:
-            self._callback_status_msg += f"\nstatus from done callback: {status_msg}"
-
-        if callback_status == LoadStatus.LOAD_SUCCESSFUL:
-            await self._send(JobState.COMPLETED)
-        else:
-            assert callback_status == LoadStatus.LOAD_FAILURE
-            await self._send(JobState.FAILED)
+        if self._scheduler._fmok_lock:
```
Comment (berland) [inline src/ert/scheduler/job.py:226]: You can do this with `async with self._scheduler._fmok_lock:`

File: src/ert/scheduler/scheduler.py [hunk 4]
```diff
@@ -105,6 +105,7 @@ def __init__(
         self._average_job_runtime: float = 0
         self._completed_jobs_num: int = 0
         self.completed_jobs: asyncio.Queue[int] = asyncio.Queue()
+        self._fmok_lock: Optional[asyncio.Lock] = None
```
Comment (berland) [inline src/ert/scheduler/scheduler.py:108]: Does this need to be `Optional`, you can set it to `asyncio.Lock()` immediately I suppose (and allowing the `async with` syntax.

**Comment by berland** [comment]
Recommend to use a longer name for this thing rather than `fmok`.

File: src/ert/scheduler/job.py (hunk 3 above)
Comment (xjules) [reply src/ert/scheduler/job.py:226]: No, I can't unfortunately. This would need to be mocked for many tests as fmok_lock can be `None`

File: src/ert/scheduler/scheduler.py (hunk 4 above)
Comment (xjules) [reply src/ert/scheduler/scheduler.py:108]: Yes, can try it, but that would alter some tests - can see how many this would be if way too many than can revert.

**Comment by xjules** [comment]
> Recommend to use a longer name for this thing rather than `fmok`.

Any suggestion? `internalization_lock`?

**Comment by xjules** [comment]
> > Recommend to use a longer name for this thing rather than `fmok`.
> 
> Any suggestion? `internalization_lock`?

changed to `self._forward_model_ok_lock`

**Review by berland — APPROVED** [review]
nice and terse!
